### ISSUE TITLE
SFM on fisheye lens with very large FOV(fov=190)

### ISSUE THREAD

**Issue by LaFeuilleMorte** [colmap/colmap#2957]
I'm trying to run colmap on images of fisheye cameras with very large fov(fov=190), But the results seemed to be very awful.

![000022](https://github.com/user-attachments/assets/8e098184-af65-4e4d-9e03-9c1e3eec8db9)

Here's my script:

colmap feature_extractor \
    --database_path=${1}/database.db \
    --image_path=${1}/images \
    --ImageReader.camera_model=OPENCV_FISHEYE \
    --ImageReader.single_camera_per_folder=1 \
    --SiftExtraction.use_gpu=true \

colmap sequential_matcher \
    --database_path=${1}/database.db \
    --SiftMatching.use_gpu=true

mkdir -p ${1}/sparse
colmap mapper \
    --database_path=${1}/database.db \
    --image_path=${1}/images \
    --output_path=${1}/sparse

**Comment by ahojnnes** [comment]
I don't think you will have success reconstructing this scene without some prior calibration of the camera and then supplying the distortion parameters to colmap.

**Comment by SkyRiderMike** [comment]
> I don't think you will have success reconstructing this scene without some prior calibration of the camera and then supplying the distortion parameters to colmap.

Could you provide any suggestions about this kind of camera ?

**Comment by sarlinpe** [comment]
Either use images from a scene with more texture, or calibrate your image manually with https://github.com/ethz-asl/kalibr for example.

**Comment by samhubs** [comment]
In addition to above, another simpler approach would be to use fiducial markers for calibration: https://docs.opencv.org/4.x/da/d13/tutorial_aruco_calibration.html. Charuco board in my experience has yielded better results.

**Comment by ishipachev** [comment]
Another option is to use OpenCV fisheye calibration algorithms with boards. As @samhubs have mentioned above.
I'm not sure if Kalibr supports that though. So the solution can be:
* Crop images to avoid having >180 setup
* Get fisheye calibration parameters
* Scale back parameters to fit initial image size (principal point)

With fisheye cameras it is important to get inital parameters.

**Comment by DuVogel87** [comment]
@LaFeuilleMorte 

Hi. Did you ever had any success. 

I found that it is very easy to use circular fisheye images with colmap.

How did you proceed?

**Comment by ahojnnes** [comment]
Closing: external calibration tools recommended for extreme fisheye lenses.
